node/sync/src/block_sync.rs

```diff
@@ -461,6 +461,9 @@ impl<N: Network> BlockSync<N> {
         let is_synced = num_blocks_behind <= max_blocks_behind;
         // Update the sync status.
         self.is_block_synced.store(is_synced, Ordering::SeqCst);
+        // Update the `IS_SYNCED` metric.
+        #[cfg(feature = "metrics")]
+        metrics::gauge(metrics::bft::IS_SYNCED, is_synced);
```

Comment:
This looks good! One thing that is a bit odd is that in our design we "register" all metrics before updating them, which basically initializes all our metrics in our names folder at 0. Its probably a trivial difference, as for this metric we initialize it and then update whether its synced or not in the node, however just wanted to point out that it will be initialized at 0. If we only want to start recording the metric the first time we hit this line of code, we should remove the metric name from the `names.rs` file

Reply:
I think it should be fine since `is_block_synced` is initialized with `Default::default() == false`. So we'll have the property that `is_block_synced` matches `IS_SYNCED`.

Reply:
That's true it'll match the value in the `BlockSync` struct but 

If a node starts off synced it's possible for the value stored in `is_block_synced` to be `false` while the node IS indeed synced. So it doesn't match the picture of what's currently synced. This might cause confusions on the Dashboards.

I'd advocate to initialize this guage as `true`/`1` because as soon as a node begins the syncing process it'll immediately set the metric to 0.

Reply:
@miazn - reading further, this does get triggered with the sync is initialized. You're right.